inc/class-sitemaps.php

```diff
@@ -282,7 +284,15 @@ function redirect( $query ) {
 			}
 
 			if ( $caching ) {
-				set_transient( 'wpseo_sitemap_cache_' . $type . '_' . $this->n, $this->sitemap, DAY_IN_SECONDS );
+				/**
+				 * We need to set a timeout, otherwise the transient is loaded every request!
+				 *
+				 * See: https://codex.wordpress.org/Function_Reference/set_transient
+				 * NB: transients that never expire are autoloaded, whereas transients with an expiration time
+				 * are not autoloaded. Consider this when adding transients that may not be needed on every
+				 * page, and thus do not need to be autoloaded, impacting page performance.
+				 */
+				set_transient( $sitemap_cache_key, $this->sitemap, DAY_IN_SECONDS );
```

Comment:
@rarst see https://github.com/Yoast/wordpress-seo/pull/4079 for the slightly modified PR for trunk.